ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/PerfSupport.scala

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.indexer.parallel
+
+import akka.stream.OverflowStrategy
+import akka.stream.scaladsl.Source
+
+object PerfSupport {
+
+  case class AverageCounter() {
```

Comment:
I would consider replacing this counter with standard Histogram. There is prior art, see the metrics called `lookup_active_contract_count_per_execution`. The benefit of using the histogram is that non-integer averages can be collected, also we will get all sorts of other useful stats

Reply:
This class is only used to feed the Gauge that displays the *current* batch size (or more precisely, the average batch size since the last gauge update).

Changing to a histogram would give you more information about the distribution of batch sizes, but it would be harder to track high-frequency fluctuations in the batch sizes, as histograms use a `ExponentiallyDecayingReservoir` by default, which "produces quantiles which are representative of (roughly) the last five minutes of data".

If we are interested in seeing detailed patterns in the fluctuations of batch sizes, then we should keep the current `AverageCounter + Gauge` combo. Otherwise, I would remove `AverageCounter` and replace the batch size measurement by a histogram, optionally switching to a sliding (time) window reservoir if we are interested in capturing high-frequency changes.